icu4c/source/common/propname.cpp

```diff
@@ -303,7 +303,10 @@ u_getPropertyEnum(const char* alias) {
 U_CAPI const char* U_EXPORT2
 u_getPropertyValueName(UProperty property,
                        int32_t value,
-                       UPropertyNameChoice nameChoice) {
+                       UPropertyNameChoice nameChoice) NO_SANITIZE_UNDEFINED {
```

Comment:
Would casting the incoming enum values to ints in the range checks work as an alternative to NO_SANITIZE_UNDEFINED, for this kind of check?

It might be slightly safer, in that it would avoid the undefined behavior, instead of suppressing checking for it. Deals with the off chance that some perverse future compiler recognizes that the if condition can only be true for undefined behavior conditions and decides to optimize the whole check away.

Reply:
good idea. Let me check would that work.

Reply:
I tried, UBSAN is not happen the moment the code access the incoming enum, in the moment if I try to cast it to int. UBSAN detect the value is out of defined values before the casting, triggered by the casting.